pkgs/development/python-modules/slither-analyzer/default.nix

```diff
@@ -46,30 +46,14 @@ buildPythonPackage rec {
       --prefix PATH : "${lib.makeBinPath [ solc ]}"
   '';
 
-  # required for pythonImportsCheck
-  postInstall = ''
-    export HOME="$TEMP"
-  '';
+  # test if the binary works
+  checkPhase = ''
+    runHook preCheck
 
-  pythonImportsCheck = [
-    "slither"
-    "slither.all_exceptions"
-    "slither.analyses"
-    "slither.core"
-    "slither.detectors"
-    "slither.exceptions"
-    "slither.formatters"
-    "slither.printers"
-    "slither.slither"
-    "slither.slithir"
-    "slither.solc_parsing"
-    "slither.utils"
-    "slither.visitors"
-    "slither.vyper_parsing"
-  ];
+    HOME="$TEMP" $out/bin/slither --version
```

Comment:
That's why we have passthru.tests.

Reply:
Pardon my ignorance of Python module packaging. In the case of pkgs/development/python-modules/slither-analyzer/default.nix, `pyproject = true` is already there. What do we need to do to make the package build fail? What I did was `nix build .#slither-analyzer`, and everything passed until I added the checkPhase.

Reply:
You build `slither-analyzer.passthru.tests`.

Reply:
If you want you can keep this but remove `passthru.tests`.

Reply:
> `pyproject = true` is already there

The other packages still use `format`.

Reply:
Cleaned up everything. Please review again.

Reply:
see https://github.com/NixOS/nixpkgs/blob/f0850972708cea377b39a03a52c8ae628868d191/doc/build-helpers/testers.chapter.md#check-a-program-version-using-all-the-default-values for help on how to do this

Reply:
I'd like to keep the `checkPhase` too though, since otherwise the build passes without warning if the binary didn't work; since the package doesn't come with a basic some test.

Reply:
fine for me